apple/internal/watchos_rules.bzl

```diff
@@ -703,6 +703,25 @@ later will be met with a rejection.
 
 Please remove the assigned watchOS 2 app `extension` and make sure a valid watchOS application
 delegate is referenced in the single-target `watchos_application`'s `deps`.
+""")
+
+    if len(ctx.attr.extensions) > 1:
+        fail("""
+Extension-based watchOS applications do not support embedded app extensions.
+
+Please remove app extensions from this target, and instead add them to the watchos_extension `extensions`.
+""")
+
+    watch_extension = None
+    if ctx.attr.extension:
+        watch_extension = ctx.attr.extension
+    elif len(ctx.attr.extensions) == 1 and ctx.attr.extensions[0][AppleBundleInfo].product_type == apple_product_type.watch2_extension:
+        watch_extension = ctx.attr.extensions[0]
+    else:
+        fail("""
+Extension-based watchOS applications require a valid watchos_extension.
+
+Please add a watchos_extension to this target `extensions`.
```

Comment:
```suggestion
Please add a `watchos_extension` to this target `extension` attribute.
```

Reply:
Notice that I'm suggesting the `extension` attribute (instead of the new plural version) since that has always been used to reference the watchOS extension (instead of the app extension), right?

Reply:
Oh I see we're marking that attribute as deprecated, so ignore the above comment and suggestion. Let's just add the formatting for the rule name.